lib/ansible/modules/network/fortios/fortios_address_group.py

```diff
@@ -0,0 +1,238 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = """
+---
+module: fortios_address_group
+version_added: "2.3"
+author: "Benjamin Jolivot (@bjolivot)"
+short_description: Manage fortios firewall address group objects
+description:
+  - This module provide management of firewall address groups on FortiOS devices.
+extends_documentation_fragment: fortios
+options:
+  name:
+    description:
+      - Name of the address group.
+    required: true
+    aliases:
+      - group_name
+  member:
+    description:
+      - Member(s) address name to add or delete.
+    required: true
+    aliases:
+      - members
+  state:
+    description:
+      - Specifies if address need to be added or deleted in group.
+    default: 'present'
+    choices: ['present', 'absent']
+  comment:
+    description:
+      - free text to describe address group.
+notes:
+  - This module requires pyFG python library.
+"""
+
+EXAMPLES = """
+- name: Add google DNS address object in dns_servers address group
+  fortios_address_group:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    state: present
+    name: dns_servers
+    member: google_dns
+    comment: "Public DNS servers"
+
+- name: Remove yahoo from trusted mail providers
+  fortios_address_group:
+    host: 192.168.0.254
+    username: admin
+    password: password
+    state: absent
+    name: trusted_mail_providers
+    member: yahoo_server
+    comment: "Trusted Email providers"
+"""
+
+RETURN = """
+firewall_address_config:
+  description: full firewall adresses config string
+  returned: always
+  type: string
+change_string:
+  description: The commands executed by the module
+  returned: only if config changed
+  type: string
+"""
+
+import re
+import shlex
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+
+from ansible.module_utils.fortios import fortios_argument_spec, fortios_required_if
+from ansible.module_utils.fortios import backup
+
+#check for pyFG lib
+try:
+    from pyFG import *
+    from pyFG.fortios import logger
+    from pyFG.exceptions import *
+    HAS_PYFG=True
+except:
+    HAS_PYFG=False
+
+
+def is_invalid_name(input_str):
+    #char must be letters, digits, - , _ , .
+    #must have between 1 and 63 chars
+    reg=re.compile(r'^[a-zA-Z0-9\.\-_]{1,63}$')
+    return not reg.match(input_str)
+
+def main():
+    #define module params
+    argument_spec = dict(
+        host        = dict(required=True ),
+        username    = dict(required=True ),
+        password    = dict(required=True, type='str', no_log=True ),
+        timeout     = dict(type='int', default=60),
+        vdom        = dict(type='str' ),
```

Comment:
host, username, password, timeout and vdom is already in fortios.py in module_utils